### DISCUSSION TITLE
Understanding Discrepancies in Screener Results

### DISCUSSION THREAD

**Discussion by Hiddenleaf07** [pkjmesra/PKScreener#215] [Q&A]
Recently, I deployed PK Screener on GitHub Codespaces. However, I've noticed slight discrepancies between its results and those shared in the public channel, even after testing post-market hours. I'm getting extra results compared to the public channel.

![PKS_X_12_33_2_21-11-24_22 23 26](https://github.com/user-attachments/assets/b9c5c27b-6fd7-44bc-b8cf-3953e73ba028)
![Screenshot_20241121_225330](https://github.com/user-attachments/assets/6c687b07-2f81-4d2b-806d-dc8dc5cc7f27)

**Comment by pkjmesra** [comment]
@Hiddenleaf07 - That's most likely because many a times for some of the tickers, we get empty datasets. So, those tickers get ignored, especially during market hours. Post market-hours, the download job runs and saves all tickers' data in a pickle, but even during that download job, if we received empty data for those tickers, the results will still vary even after running post-market-hours. This happens most frequently because we hit HTTP 429 status codes (rate-limits). From your local machine, when you request, you may receive data for those missing stocks because your IP was not rate-limited. 
I hope this helps.
